When there are already a large number of jobs in the cluster, the scheduler keeps panic  when it starts

### Description

I discovered this problem when I was trying to check whether [issue 4076](https://github.com/volcano-sh/volcano/issues/4076) has been solved.

I created some queues where the total guarantee value exceeded the total amount of cluster resources, as well as a large number of jobs. However, when starting the scheduler, the scheduler panic continuously.

It seems that the increasedDeserved surpasses the remaining causes this promblem.
``` golang

// proportion.go:237
		remaining.Sub(increasedDeserved).Add(decreasedDeserved)
```


### Steps to reproduce the issue

1. Create some queues with total guarantee resources exceed the total resources of the cluster.

My k8s cluster consists of 4 nodes, and each node has 8 CPUs for total 32 CPUs.

Then I created 3 queues. The total gurantee resources is 48 CPUs.
```
$ kubectl get queue  -o custom-columns='NAME:.metadata.name, STATE:.status.state, GURANTEE_CPU:.spec.guarantee.resource.cpu' -w
NAME       STATE    GURANTEE_CPU
default   Open     <none>
root      Open     0
test      Open     24
test1     Open     12
test2     Open     12
```
2.  Create some vcjobs and each vcjob applies for a CPU of 500m.
```
$ kubectl get vcjob | wc -l
501
```
After creation, it was observed that the number of pending jobs in the queue is still 0.
```
$ kubectl get queue  -o custom-columns='NAME:.metadata.name, INQUEUE:.status.inqueue, PENDING:.status.pending, RUNNING:.status.running, STATE:.status.state'
NAME       INQUEUE    PENDING    RUNNING    STATE
default   <none>     <none>     <none>     Open
root      <none>     <none>     <none>     Open
test      1          <none>     <none>     Open
test1     <none>     <none>     <none>     Open
test2     <none>     <none>     <none>     Open
```

3. Then Startup the vc-scheduler

After a while, it will be found that the scheduler keeps restarting.
```
$ kubectl  -n volcano-system get pod 
NAME                                   READY   STATUS             RESTARTS      AGE
volcano-admission-65bf6b7b68-8smz2     1/1     Running            0             6m7s
volcano-admission-65bf6b7b68-t4k94     1/1     Running            0             6m31s
volcano-controllers-5856757f76-5dtnz   1/1     Running            0             7m29s
volcano-controllers-5856757f76-bcf9h   1/1     Running            0             7m45s
volcano-scheduler-7fff987478-7f2tk     0/1     CrashLoopBackOff   2 (14s ago)   2m9s
volcano-scheduler-7fff987478-8gn5p     1/1     Running            3 (8s ago)    2m9s
```

The error stack in the scheduler log
```
I0509 07:09:56.012027       1 proportion.go:93] Considering Job <default/stress-job-0322-0caaa2d8-273d-4d60-afd5-e2beb837acc7>.
I0509 07:09:56.012040       1 proportion.go:93] Considering Job <default/stress-job-0275-b24cdb6a-2140-4d7c-b8bf-f017a41d7c84>.
I0509 07:09:56.012132       1 proportion.go:200] Considering Queue <test>: weight <1>, total weight <2>.
I0509 07:09:56.012167       1 proportion.go:216] Format queue <test> deserved resource to <cpu 24000.00, memory 471859200.00, pods 1.00>
I0509 07:09:56.012190       1 proportion.go:220] queue <test> is meet
I0509 07:09:56.012204       1 proportion.go:226] The attributes of queue <test> in proportion: deserved <cpu 24000.00, memory 471859200.00, pods 1.00>, realCapability <cpu 24000.00, memory 57538854912.00, attachable-volumes-csi-nfs.csi.k8s.io 8589934588.00, ephemeral-storage 745905599688000.00, pods 320.00>, allocate <cpu 0.00, memory 0.00>, request <cpu 500.00, memory 471859200.00, pods 1.00>, elastic <cpu 0.00, memory 0.00>, share <0.00>
I0509 07:09:56.012242       1 proportion.go:200] Considering Queue <test1>: weight <1>, total weight <2>.
I0509 07:09:56.012260       1 proportion.go:216] Format queue <test1> deserved resource to <cpu 12000.00, memory 4294967296.00, pods 60.00>
I0509 07:09:56.012274       1 proportion.go:220] queue <test1> is meet
I0509 07:09:56.012284       1 proportion.go:226] The attributes of queue <test1> in proportion: deserved <cpu 12000.00, memory 4294967296.00, pods 60.00>, realCapability <cpu 12000.00, memory 61833822208.00, ephemeral-storage 745905599688000.00, pods 380.00, attachable-volumes-csi-nfs.csi.k8s.io 8589934588.00>, allocate <cpu 0.00, memory 0.00>, request <cpu 0.00, memory 0.00>, elastic <cpu 0.00, memory 0.00>, share <0.00>
I0509 07:09:56.012333       1 panic.go:770] End scheduling ...
E0509 07:09:56.012477       1 assert.go:33] "Observed a panic" panic="resource is not sufficient to do operation: <cpu 32000.00, memory 66128789504.00, hugepages-1Gi 0.00, hugepages-2Mi 0.00, pods 440.00, attachable-volumes-csi-nfs.csi.k8s.io 8589934588.00, ephemeral-storage 745905599688000.00> sub <cpu 36000.00, memory 4766826496.00, pods 61.00>" stacktrace=<
	goroutine 369 [running]:
	k8s.io/apimachinery/pkg/util/runtime.logPanic({0x28f8478, 0x3ef7500}, {0x206a720, 0xc0009054f0})
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:107 +0xbc
	k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x28f8478, 0x3ef7500}, {0x206a720, 0xc0009054f0}, {0x3ef7500, 0x0, 0x100000000000000?})
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:82 +0x5e
	k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x12?})
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:59 +0x108
	panic({0x206a720?, 0xc0009054f0?})
		/usr/local/go/src/runtime/panic.go:770 +0x132
	volcano.sh/volcano/pkg/scheduler/util/assert.Assert(0x10?, {0xc00046a480?, 0xc0008194f8?})
		/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:33 +0x165
	volcano.sh/volcano/pkg/scheduler/util/assert.Assertf(0x0, {0x2607110?, 0x1?}, {0xc0008194f8?, 0xc000819528?, 0x83cf1f?})
		/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:43 +0x4a
	volcano.sh/volcano/pkg/scheduler/api.(*Resource).Sub(0xc000379080, 0xc000379280)
		/go/src/volcano.sh/volcano/pkg/scheduler/api/resource_info.go:260 +0x90
	volcano.sh/volcano/pkg/scheduler/plugins/proportion.(*proportionPlugin).OnSessionOpen(0xc0014d8480, 0xc0001b7000)
		/go/src/volcano.sh/volcano/pkg/scheduler/plugins/proportion/proportion.go:237 +0x14a5
	volcano.sh/volcano/pkg/scheduler/framework.OpenSession({0x2911898?, 0xc000352b08?}, {0xc000542270, 0x2, 0x2}, {0x0, 0x0, 0x0})
		/go/src/volcano.sh/volcano/pkg/scheduler/framework/framework.go:45 +0x312
	volcano.sh/volcano/pkg/scheduler.(*Scheduler).runOnce(0xc000372600)
		/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:117 +0x29d
	k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x33
	k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000af5b20, {0x28d6580, 0xc000c66840}, 0x1, 0xc000524a80)
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xaf
	k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000af5b20, 0x3b9aca00, 0x0, 0x1, 0xc000524a80)
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x7f
	k8s.io/apimachinery/pkg/util/wait.Until(...)
		/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161
	created by volcano.sh/volcano/pkg/scheduler.(*Scheduler).Run in goroutine 107
		/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:91 +0x1a7
 >
panic: resource is not sufficient to do operation: <cpu 32000.00, memory 66128789504.00, hugepages-1Gi 0.00, hugepages-2Mi 0.00, pods 440.00, attachable-volumes-csi-nfs.csi.k8s.io 8589934588.00, ephemeral-storage 745905599688000.00> sub <cpu 36000.00, memory 4766826496.00, pods 61.00> [recovered]
	panic: resource is not sufficient to do operation: <cpu 32000.00, memory 66128789504.00, hugepages-1Gi 0.00, hugepages-2Mi 0.00, pods 440.00, attachable-volumes-csi-nfs.csi.k8s.io 8589934588.00, ephemeral-storage 745905599688000.00> sub <cpu 36000.00, memory 4766826496.00, pods 61.00>

goroutine 369 [running]:
k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x28f8478, 0x3ef7500}, {0x206a720, 0xc0009054f0}, {0x3ef7500, 0x0, 0x100000000000000?})
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:89 +0xee
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x12?})
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:59 +0x108
panic({0x206a720?, 0xc0009054f0?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
volcano.sh/volcano/pkg/scheduler/util/assert.Assert(0x10?, {0xc00046a480?, 0xc0008194f8?})
	/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:33 +0x165
volcano.sh/volcano/pkg/scheduler/util/assert.Assertf(0x0, {0x2607110?, 0x1?}, {0xc0008194f8?, 0xc000819528?, 0x83cf1f?})
	/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:43 +0x4a
volcano.sh/volcano/pkg/scheduler/api.(*Resource).Sub(0xc000379080, 0xc000379280)
	/go/src/volcano.sh/volcano/pkg/scheduler/api/resource_info.go:260 +0x90
volcano.sh/volcano/pkg/scheduler/plugins/proportion.(*proportionPlugin).OnSessionOpen(0xc0014d8480, 0xc0001b7000)
	/go/src/volcano.sh/volcano/pkg/scheduler/plugins/proportion/proportion.go:237 +0x14a5
volcano.sh/volcano/pkg/scheduler/framework.OpenSession({0x2911898?, 0xc000352b08?}, {0xc000542270, 0x2, 0x2}, {0x0, 0x0, 0x0})
	/go/src/volcano.sh/volcano/pkg/scheduler/framework/framework.go:45 +0x312
volcano.sh/volcano/pkg/scheduler.(*Scheduler).runOnce(0xc000372600)
	/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:117 +0x29d
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000af5b20, {0x28d6580, 0xc000c66840}, 0x1, 0xc000524a80)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xaf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000af5b20, 0x3b9aca00, 0x0, 0x1, 0xc000524a80)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x7f
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161
created by volcano.sh/volcano/pkg/scheduler.(*Scheduler).Run in goroutine 107
	/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:91 +0x1a7
```




### Describe the results you received and expected

Why does this problem exist and how can I fix it.

### What version of Volcano are you using?

v1.11.2

### Any other relevant information

k8s: 1.28.2
os: centos7

--- a/pkg/scheduler/plugins/proportion/proportion.go
+++ b/pkg/scheduler/plugins/proportion/proportion.go
@@ -234,7 +234,7 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 			metrics.UpdateQueueDeserved(attr.name, attr.deserved.MilliCPU, attr.deserved.Memory, attr.deserved.ScalarResources)
 		}
 
-		remaining.Sub(increasedDeserved).Add(decreasedDeserved)
+		remaining = api.ExceededPart(remaining.Clone().Add(decreasedDeserved), increasedDeserved)
 		klog.V(4).Infof("Remaining resource is  <%s>", remaining)
 		if remaining.IsEmpty() || equality.Semantic.DeepEqual(remaining, oldRemaining) {
 			klog.V(4).Infof("Exiting when remaining is empty or no queue has more resource request:  <%v>", remaining)

